nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -2144,29 +2002,29 @@ static int nxagentGetDialogName()
 {
   strcpy(nxagentDialogName, "NX");
 
-  *(nxagentDialogName + 255) = '\0';
+  nxagentDialogName[sizeof(nxagentDialogName) - 1] = '\0';
 
   if (*nxagentSessionId != '\0')
   {
-    int length = strlen(nxagentSessionId);
+    int length = strlen(nxagentSessionId) - (MD5_LENGTH * 2 + 1);
+    const char prefix[] = "NX - ";
 
-    strcpy(nxagentDialogName, "NX - ");
+    strcpy(nxagentDialogName, prefix);
 
-    if (length > (MD5_LENGTH * 2 + 1) &&
-           *(nxagentSessionId + (length - (MD5_LENGTH * 2 + 1))) == '-')
+    if (length > 0 && nxagentSessionId[length] == '-')
     {
-      strncat(nxagentDialogName, nxagentSessionId, length - (MD5_LENGTH * 2 + 1));
+      strncat(nxagentDialogName, nxagentSessionId,
+              MIN(length, sizeof(nxagentDialogName) - 1 - strlen(prefix)));
```

Comment:
Same problem as above.

Bounding the size is actually a pretty good idea here, though.